azext_iot/operations/_mqtt.py

```diff
@@ -5,92 +5,87 @@
-    2: "refused - invalid client id",
-    3: "refused - server unavailable",
-    4: "refused - bad username or password",
-    5: "refused - not authorized",
-}
-
-
-class mqtt_client_wrap(object):
-    def __init__(self, target, device_id, properties=None, sas_duration=3600):
-        self.target = target
+class mqtt_client(object):
+    def __init__(self, target, device_conn_string, device_id, method_response_code=None, method_response_payload=None):
         self.device_id = device_id
-
-        sas = SasTokenAuthentication(
-            target["entity"],
-            target["policy"],
-            target["primarykey"],
-            sas_duration,
-        ).generate_sas_token()
-        cwd = EXTENSION_ROOT
-        cert_path = os.path.join(cwd, "digicert.pem")
-        tls = {"ca_certs": cert_path, "tls_version": ssl.PROTOCOL_SSLv23}
-        self.topic_publish = "devices/{}/messages/events/{}".format(
-            device_id, url_encode_dict(properties) if properties else ""
-        )
-        self.topic_receive = "devices/{}/messages/devicebound/#".format(device_id)
-        self.connected = False
-
-        self.client = mqtt.Client(protocol=mqtt.MQTTv311, client_id=device_id)
-        self.client.on_connect = self.on_connect
-        self.client.on_message = self.on_message
-        self.client.on_publish = self.on_publish
-        self.client.tls_set(ca_certs=tls["ca_certs"], tls_version=tls["tls_version"])
-        self.client.username_pw_set(username=build_mqtt_device_username(target["entity"], device_id), password=sas)
-        self.client.connect(host=self.target["entity"], port=8883)
-
-    def on_connect(self, client, userdata, flags, rc):
-        six.print_(
-            "Connected to target IoT Hub MQTT broker with result: {}".format(
-                connection_result[rc]
-            )
-        )
-        self.client.subscribe(self.topic_receive)
-        six.print_("Subscribed to device bound message queue")
-        self.connected = True
-
-    def on_message(self, client, userdata, msg):
+        self.device_client = mqtt_device_client.create_from_connection_string(device_conn_string)
+        self.device_client.connect()
+        self.device_client.on_message_received = self.message_handler
+        self.device_client.on_method_request_received = self.method_request_handler
+        self.method_response_code = method_response_code
+        self.method_response_payload = method_response_payload
+
+    def send_d2c_message(self, message_text, properties=None):
+        message = Message(message_text)
+        message.custom_properties = properties
+        self.device_client.send_message(message)
+
+    def message_handler(self, message):
         six.print_()
-        six.print_("_Received C2D message with topic_: {}".format(msg.topic))
-        six.print_("_Payload_: {}".format(msg.payload))
+        six.print_("_Received C2D message with topic_: /devices/{}/messages/devicebound".format(self.device_id))
```

Comment:
All references to `six` should be removed. But we can get to it in the next iteration.

Reply:
Yeah I will address this in the next iteration